File: src/manager/parser/ParserManagerImpl.java
package manager.parser;

import com.google.gson.Gson;

import java.lang.reflect.Type;

public class ParserManagerImpl implements ParserManager {
    private Gson mGson;

    public ParserManagerImpl(Gson gson) {
        mGson = gson;
    }

    public String toJson(Object object) {
        return mGson.toJson(object);
    }

    public <T> T fromJson(String json, Class<T> type) {
        return mGson.fromJson(json, type);
    }

    @Override
    public <T> T fromJson(String json, Type type) {
        return mGson.fromJson(json, type);
    }
}

File: resources/BaseProjectKotlin/app/src/main/java/co/bybardo/myapp/infrastructure/manager/serialization/ParserManagerImpl.kt
/*
 * Created by Julián Falcionelli on 2019.
 * Copyright © 2019 Bardo (bybardo.co). All rights reserved.
 * Happy Coding !
 */

package co.bybardo.myapp.infrastructure.manager.serialization

import com.google.gson.Gson
import java.lang.reflect.Type

class ParserManagerImpl(private val gson: Gson) : ParserManager {

    override fun toJson(`object`: Any): String {
        return gson.toJson(`object`)
    }

    override fun <T> fromJson(json: String, type: Class<T>): T {
        return gson.fromJson(json, type)
    }

    override fun <T> fromJson(json: String, type: Type): T {
        return gson.fromJson<T>(json, type)
    }
}